src/storage/src/source/postgres.rs

```diff
@@ -193,35 +192,30 @@ impl SourceRender for PostgresSourceConnection {
         // restart we are going to set the status to an ssh error correctly, so we don't do this
         // extra work.
 
-        let health = snapshot_err.concat(&repl_err).flat_map(move |err| {
+        let health = snapshot_err.concat(&repl_err).map(move |err| {
             // This update will cause the dataflow to restart
             let err_string = err.display_with_causes().to_string();
             let update = HealthStatusUpdate::halting(err_string.clone(), None);
-            let namespace = if matches!(
-                &*err,
-                TransientError::PostgresError(PostgresError::Ssh(_))
-                    | TransientError::PostgresError(PostgresError::SshIo(_))
-            ) {
-                StatusNamespace::Ssh
-            } else {
-                Self::STATUS_NAMESPACE.clone()
+
+            let namespace = match err {
+                // Is there a better way to reach into an Rc?
+                ReplicationError::Transient(err)
+                    if matches!(
+                        &*err,
+                        TransientError::PostgresError(PostgresError::Ssh(_))
+                            | TransientError::PostgresError(PostgresError::SshIo(_))
+                    ) =>
+                {
```

Comment:
There isn't a way of pattern matching inside smart pointers sadly :( 

The only other alternative that avoids the macro is the following:
```rust
                ReplicationError::Transient(err) => match *err {
                    TransientError::PostgresError(PostgresError::Ssh(_) | PostgresError::SshIo(_) )  => StatusNamespace::Ssh,
                     _ =>  Self::STATUS_NAMESPACE,
                },
                _ =>  Self::STATUS_NAMESPACE,
```

Reply:
Makes sense! Realizing this shouldn't be left as a permanent comment here but not worth the $5 of another CI run––will clean up next time I'm in CI.